pootle/apps/pootle_misc/version.py

```diff
@@ -0,0 +1,150 @@
+    candidate_pos = _get_candidate_pos(version)
+    candidate = version[candidate_pos]
+    candidate_extra = version[candidate_pos+1]
+
+    sub = ''
+    if _is_development_candidate(version):
+        git_changeset = get_git_changeset()
+        if git_changeset:
+            sub = '.dev%s' % git_changeset
+
+    elif candidate != 'final':
+        mapping = {'alpha': 'a', 'beta': 'b', 'rc': 'rc'}
+        sub = mapping[candidate] + str(candidate_extra)
+
+    return str(main + sub)
+
+
+def _is_development_candidate(version):
+    candidate_pos = _get_candidate_pos(version)
+    candidate = version[candidate_pos]
+    candidate_extra = version[candidate_pos+1]
+    return candidate == 'alpha' and candidate_extra == 0
+
+
+def _get_candidate_pos(version):
+    """Returns the position of the candidate marker."""
+    return [i for i, part in enumerate(version)
+            if part in CANDIDATE_MARKERS][0]
+
+
+def _get_candidate(version):
+    """Returns the candidate. One of alpha, beta, rc or final."""
+    return version[_get_candidate_pos(version)]
+
+
+def _get_version_string(parts):
+    """Return a X.Y.Z version from the list pf version parts."""
+    return '.'.join(str(x) for x in parts)
+
+
+def get_main_version(version=None):
+    "Returns main version (X.Y[.Z]) from VERSION."
+    version = get_complete_version(version)
+    candidate_pos = _get_candidate_pos(version)
+    return _get_version_string(version[:candidate_pos])
+
+
+def get_major_minor_version(version=None):
+    "Returns X.Y from VERSION."
+    version = get_complete_version(version)
+    return _get_version_string(version[:2])
+
+
+def get_complete_version(version=None):
+    """Returns a tuple of the Pootle version. Or the supplied ``version``"""
+    if version is None:
+        from pootle import VERSION as version
+
+    return version
+
+
+def get_docs_version(version=None, positions=2):
+    """Return the version used in documentation."""
+    version = get_complete_version(version)
+    candidate = _get_candidate(version)
+    candidate_pos = _get_candidate_pos(version)
+    if positions > candidate_pos:
+        positions = candidate_pos
+    if _is_development_candidate(version):
+        return 'dev'
+    else:
+        return _get_version_string(version[:positions])
+
+
+def _get_repo_dir():
+    return os.path.dirname(os.path.abspath(__file__))
+
+
+@lru_cache()
+def _shell_command(command):
+    command_subprocess = subprocess.Popen(
+        command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True,
+        cwd=_get_repo_dir(), universal_newlines=True
+    )
+    return command_subprocess.communicate()[0]
+
+
+def get_git_changeset():
+    """Returns a numeric identifier of the latest git changeset.
+
+    The result is the UTC timestamp of the changeset in YYYYMMDDHHMMSS format.
+    This value isn't guaranteed to be unique, but collisions are very unlikely,
+    so it's sufficient for generating the development version numbers.
```

Comment:
The hash function appears later.  I'd prefer to not use that in any version output as its not sortable.